### ISSUE TITLE
citizenships

### ISSUE THREAD

**Issue by andneuma** [feminizidmap/feminizidmap-backend#26]
@lislis folgende probleme / phänomene treten hier auf

- [x] `citizenship` hab ich jetzt, mach das wie an anderen stellen auch über string matching und mapping der englischen auf die deutschen namen. glaub damit kommen wir gut durch
- [x] `citizenship_type` in Excel scheint auch erstmal okay zu mappen auf die Dropdowns
- [ ] Bei `foreign_citizenship_type` haben wir das Problem, dass die Values in Excel (weiß nicht wie flächendeckend) nicht so richtig mappen auf die Dropdown Values. E.g. "Excel: Aufenthaltsgenehmigung (ohne Geflüchtete)" => "Strapi: ?"
- [ ] Auch gibts das Problem dass im Dropdown hier zwei Values sind die zusätzliche angaben benötigen ("Ja" und "Sonstige"). D.h. eigentlich braucht es streng genommen noch ein weiteres Feld wo die Details aufgenommen werden
* Frage hier wäre auch, welche Values / Cases von `citizenship` und `citizenship_type` auf welche Values in `foreign_citizenship_type` mappen? Vlt kannst du das der Einfachheit halber hier in dieser Liste abchecken und weitere Fälle ergänzen
  - [ ] **n/a** + **nil** in `foreign_citizenship_type` mappen auf `nil`
  - [ ] **entfällt** in `foreign_citizenship_type` mappen auf `entfällt`
  - [ ] **Deutsche Staatsbürgerschaft** in `citizenship_type`  mappt auf `entfällt`
  - [ ] **Ausländische Staatsbürgerschaft** in `citizenship_type` mappt auf `Ja` 
  - [ ] **Doppelte Staatsbürgerschaft** in `citizenship_type` mappt auf `Ja`
- [ ] Das mit der doppelten Staatsbürgerschaft ist mir auch noch nicht so ganz klar. Gibts dafür ein einheitliches Format der Eintragung in den Excel-Daten?

Und noch ein Metapunkt zum Thema Migrantisierung: Wie sind da eure Diskussionen? Im Dropdown werden IMO "Staatsbürgerschaft" und "Migrationsgeschichte" + Zuschreibungen dritter ("Migrantisierung") gemischt, die Relation ist auch nicht auf mehrfachnennung ausgelegt. Oder versteh ich das falsch?

**Comment by lislis** [comment]
Ich versuch mal das mapping zu erklären, weil ich das auf Strapi haben wollte, was schon anders ist als im Excel. Und meine Benennungen nicht unbedingt Sinn ergeben. Sorry für Wiederholungen von deinem Post, aber ich muss mich da jetzt selbst nochmal durchdenken.

- `Excel:Staatsbürgerschaft` mappt auf  `strapi:citizenship_type` 
- `Excel:Ausländische Staatsbürgerschaft`  habe ich die Exceldropdown-Optionen aufgeteilt in: 
    - `strapi:citizenship` als Länderdropdown
    - `strapi:foreign_citizenship_type` für die Optionen "Migrationshintergrund" etc
- `Excel:Rechtlicher Status` mapped auf `strapi:legal_status`

Jetzt wo ich so drüber schreibe, ist es eigentlich nicht ganz so kompliziert. Die wichtigen Infos sind:
- Deutsche, ausländische oder Doppelte Staatsbürgerschaft?
    - wenn ausländische, welche? 
    - wenn doppelte, welche? 
- Gibt es Migrationshintergrund?
- Sonstiges Kommentarfeld für diesen Bereich (Das Feld mach ich noch, ist bis jetzt nicht da)

Ich stimm dir voll zu, dass die derzeitige Aufteilung eine seltsame Mischung ist, und auch das Format im Excel nicht unbedingt Sinn macht. So richtig diskutiert wurde das meiner Erinnerung nach nicht, aber ich kann da nochmal nachfragen.

Eigentlich macht es dann Sinn das feld `foreign_citizenship_type` umzubenennen in `migration_background` und das als Ja/Nein/Entfällt Feld zu machen, plus das Kommentarfeld.

Damit könntest du dann aus dem `Excel:Ausländische Staatsbürgerschaft` Option "Migrationshintergrund" -> `strapi:migration_background` Option "Ja" machen. Und alles andere würde ich in `strapi:migration_background` als nil mappen, weil wir die Infos in der Auflösung nicht haben.

Mit der Doppelten Staatsbürgerschaft gibt es nicht so viele Beispiele, aber ich würde die andere Staatsbürgerschaft dann auch in `citizenship` festhalten. Meintest du das?

**Comment by andneuma** [comment]
> Eigentlich macht es dann Sinn das feld foreign_citizenship_type umzubenennen in migration_background und das als Ja/Nein/Entfällt Feld zu machen, plus das Kommentarfeld.

Genau, hier wäre meine Frage konkret, ob man das aus den Daten rauslesen kann? Also wenn jemand kosovarische Staatsbürger*innenschaft hat, dann würde ich das auf "Ja" mappen, wenn deutsche Staatsbürger*innenschaft weil hier geboren aber in den älteren Generationen es eine Migrationsgeschichte (Bsp vietnamesiche Gastarbeiter*innen in der DDR gewesen und geblieben) gibt dann wird es schwieriger oder? Also Frage anders formuliert ist "wo beginnt Migrationshintergrund und wo hört er auf", etc. Is voll kompliziert und ja auch ne soziologische Disziplin. Ich wills auch nicht unnötig verkomplizieren. Dachte nur da der Anspruch an die Daten ja ein akademischer ist macht es sicherlich Sinn, dass das Mapping am Ende akkurat ist.

**Comment by lislis** [comment]
hmm, puh. Ich hab nochmal im Review-Protokoll nachgelesen dazu, und scheinbar kam das ursprünglich daher, das wenn in Medienberichten sowas wie "mit xyzen Wurzeln" oder "xyzer Abstammung" steht, das irgendwie abzubilden. Also irgendwie gar nicht so super tief gedacht. 
Deswegen würde ich ältere Generationen ausklammern und das nur auf die aktuelle Person beziehen. Aber ich nehm das nochmal mit ins Plenum

**Comment by lislis** [comment]
Soooo, wir haben das gestern ein bisschen diskutiert und der Stand ist folgender:
- die ganze migrations-hintergrundsgeschichte wird erstmal rausgenommen, 3/4 der leute wussten auch nicht, dass "migrantisiert" als Option in dem einen Dropdown im Excel überhaupt existiert
- weil es aber schon als datenpunkt interessant ist, soll es aber irgendwann doch mit erhoben werden, vor allem wie medien darüber berichten. Aber, das erfordert noch viel mehr diskussionen über wie genau. Deswegen jetzt erstmal ausklammern

Was heißt das für den Import? 
- Excel:Staatsbürgerschaft mappt auf strapi:citizenship_type
- Excel:Ausländische Staatsbürgerschaft mappt auf strapi:citizenship
- if Excel:Ausländische Staatsbürgerschaft == "Migrantisiert" then add "Migrantisiert" zu strapi:citizenship_details

@andneuma klingt das gut?

**Comment by andneuma** [comment]
> Was heißt das für den Import?

    Excel:Staatsbürgerschaft mappt auf strapi:citizenship_type
    Excel:Ausländische Staatsbürgerschaft mappt auf strapi:citizenship
    if Excel:Ausländische Staatsbürgerschaft == "Migrantisiert" then add "Migrantisiert" zu strapi:citizenship_details

**Comment by andneuma** [comment]
> Was heißt das für den Import?

    Excel:Staatsbürgerschaft mappt auf strapi:citizenship_type
    Excel:Ausländische Staatsbürgerschaft mappt auf strapi:citizenship
    if Excel:Ausländische Staatsbürgerschaft == "Migrantisiert" then add "Migrantisiert" zu strapi:citizenship_details

@lislis das gilt auch für die perpetrators, right?

**Comment by lislis** [comment]
If this is in the import script, this can be closed, too!
